Sources/Host/POSIX/File.cpp

```diff
@@ -54,5 +55,28 @@ ErrorCode File::pread(ByteVector &buf, uint64_t &count, uint64_t offset) {
 
   return _lastError = kSuccess;
 }
+
+ErrorCode File::createDirectory(std::string const &path, uint32_t flags) {
+  size_t pos = 0;
+
+  // Each parent directory must be created individually, if it does not exist
+  while ((pos = path.find('/', pos + 1)) != std::string::npos) {
+    std::string partialPath = path.substr(0, pos);
+
+    if (!Platform::IsFilePresent(partialPath.c_str())) {
+      if (::mkdir(partialPath.c_str(), flags) < 0) {
+        return Platform::TranslateError();
+      }
+    }
+  }
+
+  if (!Platform::IsFilePresent(path.c_str())) {
```

Comment:
I think if you don't do a `for` loop but keep a `while` loop, you can do a `do {} while();` loop to avoid doing this code twice.